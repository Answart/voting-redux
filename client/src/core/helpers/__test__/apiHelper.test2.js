import { requestOpts, requestApi, handleResponse, createHeaders } from '../api.helper';
import nock from 'nock';


describe('api.helper', () => {
  describe('requestOpts()', () => {

    it('', () => {

    });
  });

  describe('requestApi()', () => {

    it('', () => {

    });
  });

  describe('handleResponse()', () => {

    it('', () => {

    });
  });

  describe('createRequest()', () => {

    it('', () => {

    });
  });
});



test('method defaults to GET', t => {
  const reply = { foo: 'bar' };
  nock(API_URL)
    .get('/foo')
    .reply(200, reply);
  return callApi('foo').then(response => {
    t.deepEqual(response, reply);
  });
});

test('sends the body', t => {
  const body = { id: 5 };
  const reply = { foo: 'bar' };
  nock(API_URL)
    .post('/foo', body)
    .reply(200, reply);
  return callApi('foo', 'post', body).then(response => {
    t.deepEqual(response, reply);
  });
});

test('returns the error', t => {
  const reply = { message: 'Errrrrrrrrr' };
  nock(API_URL)
    .get('/send_error')
    .reply(500, reply);
  return callApi('send_error').then(error => {
    t.deepEqual(error, reply);
  });
});





import { API_TRACKS_URL, API_USERS_URL, CLIENT_ID_PARAM, PAGINATION_PARAMS } from 'src/core/constants';
import { api, dispatch, requestUrl } from '../api-service';


describe('api', () => {
  describe('service', () => {
    let body;
    let queryKey;
    let queryValue;
    let queryParam;
    let server;
    let successResponse;


    beforeEach(() => {
      body = {};
      queryKey = 'q';
      queryValue = 'test';
      queryParam = `${queryKey}=${queryValue}`;
      // server = sinon.fakeServer.create();
      // successResponse = [200, {'Content-type': 'application/json'}, JSON.stringify(body)];
    });


    //afterEach(() => {
      // server.restore();
    //});


    describe('requestUrl()', () => {
      it('should add client id param to url', () => {
        let url = requestUrl({url: API_TRACKS_URL});
        expect(url).toMatch(CLIENT_ID_PARAM);
      });

      it('should NOT add duplicate client id param to url', () => {
        let regExp = new RegExp(CLIENT_ID_PARAM, 'gi');
        let url = requestUrl({url: `${API_TRACKS_URL}?${CLIENT_ID_PARAM}`});
        expect(url.match(regExp).length).toBe(1);
      });

      it('should add pagination params to url if options.paginate is true', () => {
        let url = requestUrl({paginate: true, url: API_TRACKS_URL});
        expect(url).toMatch(PAGINATION_PARAMS);
      });

      it('should NOT add pagination params to url by default', () => {
        let url = requestUrl({url: API_TRACKS_URL});
        expect(url).not.toMatch(PAGINATION_PARAMS);
      });

      it('should add query params to url if options.query is provided', () => {
        let url = requestUrl({query: queryParam, url: API_TRACKS_URL});
        expect(url).toMatch(queryParam);
      });
    });


    describe('dispatch()', () => {
      const url = `${API_TRACKS_URL}?${CLIENT_ID_PARAM}`;

      xit('should set request header `Accept: application/json`', () => {
        server.respondWith('get', url, ({requestHeaders}) => {
          expect(requestHeaders.Accept).toBe('application/json');
          return successResponse;
        });

        dispatch({url: API_TRACKS_URL});
        server.respond();
      });

      xit('should resolve promise with response body', done => {
        server.respondWith('get', url, successResponse);

        dispatch({url: API_TRACKS_URL})
          .then(responseBody => {
            expect(responseBody).toEqual(body);
            done();
          })
          .catch(error => {
            fail(error);
            done();
          });

        server.respond();
      });

      xit('should reject promise for soundcloud error codes', () => {
        [400, 401, 403, 404, 406, 422, 429, 500, 503, 504]
          .forEach(code => {
            server.respondWith([code, {}, JSON.stringify({})]);

            dispatch({url: API_TRACKS_URL})
              .catch(error => {
                expect(error.status).toBe(code);
              });

            server.respond();
          });
      });
    });


    describe('api.fetch()', () => {
      xit('should perform GET request with correct url', () => {
        const url = `${API_TRACKS_URL}?${CLIENT_ID_PARAM}`;

        server.respondWith('get', url, request => {
          expect(request.url).toBe(url);
          return successResponse;
        });

        api.fetch(url);
        server.respond();
      });
    });


    describe('api.fetchSearchResults()', () => {
      xit('should perform GET request with correct url', () => {
        const url = `${API_TRACKS_URL}?${CLIENT_ID_PARAM}&${PAGINATION_PARAMS}&${queryParam}`;

        server.respondWith('get', url, request => {
          expect(request.url).toBe(url);
          return successResponse;
        });

        api.fetchSearchResults(queryValue);
        server.respond();
      });
    });


    describe('api.fetchUser()', () => {
      xit('should perform GET request with correct url', () => {
        const userId = 123;
        const url = `${API_USERS_URL}/${userId}?${CLIENT_ID_PARAM}`;

        server.respondWith('get', url, request => {
          expect(request.url).toBe(url);
          return successResponse;
        });

        api.fetchUser(userId);
        server.respond();
      });
    });


    describe('api.fetchUserLikes()', () => {
      xit('should perform GET request with correct url', () => {
        const userId = 123;
        const url = `${API_USERS_URL}/${userId}/favorites?${CLIENT_ID_PARAM}&${PAGINATION_PARAMS}`;

        server.respondWith('get', url, request => {
          expect(request.url).toBe(url);
          return successResponse;
        });

        api.fetchUserLikes(userId);
        server.respond();
      });
    });


    describe('api.fetchUserTracks()', () => {
      xit('should perform GET request with correct url', () => {
        const userId = 123;
        const url = `${API_USERS_URL}/${userId}/tracks?${CLIENT_ID_PARAM}&${PAGINATION_PARAMS}`;

        server.respondWith('get', url, request => {
          expect(request.url).toBe(url);
          return successResponse;
        });

        api.fetchUserTracks(userId);
        server.respond();
      });
    });
  });
});



import test from 'ava';
import callApi, { API_URL } from '../apiCaller';
import nock from 'nock';

test('method defaults to GET', t => {
  const reply = { foo: 'bar' };
  nock(API_URL)
    .get('/foo')
    .reply(200, reply);
  return callApi('foo').then(response => {
    t.deepEqual(response, reply);
  });
});

test('sends the body', t => {
  const body = { id: 5 };
  const reply = { foo: 'bar' };
  nock(API_URL)
    .post('/foo', body)
    .reply(200, reply);
  return callApi('foo', 'post', body).then(response => {
    t.deepEqual(response, reply);
  });
});

test('returns the error', t => {
  const reply = { message: 'Errrrrrrrrr' };
  nock(API_URL)
    .get('/send_error')
    .reply(500, reply);
  return callApi('send_error').then(error => {
    t.deepEqual(error, reply);
  });
});


// actions_spec.js

var axios = require('axios');
var Q = require('q');
var _ = require('lodash');

describe('.fetchData', function() {
	var dispatch,
	    deferred;

	beforeEach(function() {
		deferred = Q.defer();
		spyOn(axios, 'get').and.returnValue(deferred.promise);
		dispatch = jasmine.createSpy();
	});


	describe('with a successful response', function() {
		it('dispatches receiveData', function(done) {
			fetchData()(dispatch);
			var response = jasmine.createSpyObj('response', ['data']);
			deferred.resolve(response);
			_.defer(function() {
				expect(dispatch).toHaveBeenCalledWith({
					type: 'RECEIVE_DATA',
					data: response
				});
				done();
			});
		});
	});
});


// Testing your reducer alongside your actions with redux-mock-store is as simple as running your reducer function on the state that you pass to the mock store and the action you expect:

import actions from './actions'
import reducer from './reducer'
import initialState from './initialState'

const state = { ...initialState, numbers: '23' }
const expectedAction = { type: 'DELETE_LAST_NUMBER', payload: '*' }
return store.dispatch(actions.pressKey('*')).then(() => {
  expect(store.getActions()[0]).toEqual(expectedAction)
  expect(reducer(state, expectedAction)).toEqual({ ...state, numbers: '2' })
})



import { API_TRACKS_URL, API_USERS_URL, CLIENT_ID_PARAM, PAGINATION_PARAMS } from 'src/core/constants';
import { api, dispatch, requestUrl } from '../api-service';


describe('api', () => {
  describe('service', () => {
    let body;
    let queryKey;
    let queryValue;
    let queryParam;
    let server;
    let successResponse;


    beforeEach(() => {
      body = {};
      queryKey = 'q';
      queryValue = 'test';
      queryParam = `${queryKey}=${queryValue}`;
      // server = sinon.fakeServer.create();
      // successResponse = [200, {'Content-type': 'application/json'}, JSON.stringify(body)];
    });


    //afterEach(() => {
      // server.restore();
    //});


    describe('requestUrl()', () => {
      it('should add client id param to url', () => {
        let url = requestUrl({url: API_TRACKS_URL});
        expect(url).toMatch(CLIENT_ID_PARAM);
      });

      it('should NOT add duplicate client id param to url', () => {
        let regExp = new RegExp(CLIENT_ID_PARAM, 'gi');
        let url = requestUrl({url: `${API_TRACKS_URL}?${CLIENT_ID_PARAM}`});
        expect(url.match(regExp).length).toBe(1);
      });

      it('should add pagination params to url if options.paginate is true', () => {
        let url = requestUrl({paginate: true, url: API_TRACKS_URL});
        expect(url).toMatch(PAGINATION_PARAMS);
      });

      it('should NOT add pagination params to url by default', () => {
        let url = requestUrl({url: API_TRACKS_URL});
        expect(url).not.toMatch(PAGINATION_PARAMS);
      });

      it('should add query params to url if options.query is provided', () => {
        let url = requestUrl({query: queryParam, url: API_TRACKS_URL});
        expect(url).toMatch(queryParam);
      });
    });


    describe('dispatch()', () => {
      const url = `${API_TRACKS_URL}?${CLIENT_ID_PARAM}`;

      xit('should set request header `Accept: application/json`', () => {
        server.respondWith('get', url, ({requestHeaders}) => {
          expect(requestHeaders.Accept).toBe('application/json');
          return successResponse;
        });

        dispatch({url: API_TRACKS_URL});
        server.respond();
      });

      xit('should resolve promise with response body', done => {
        server.respondWith('get', url, successResponse);

        dispatch({url: API_TRACKS_URL})
          .then(responseBody => {
            expect(responseBody).toEqual(body);
            done();
          })
          .catch(error => {
            fail(error);
            done();
          });

        server.respond();
      });

      xit('should reject promise for soundcloud error codes', () => {
        [400, 401, 403, 404, 406, 422, 429, 500, 503, 504]
          .forEach(code => {
            server.respondWith([code, {}, JSON.stringify({})]);

            dispatch({url: API_TRACKS_URL})
              .catch(error => {
                expect(error.status).toBe(code);
              });

            server.respond();
          });
      });
    });


    describe('api.fetch()', () => {
      xit('should perform GET request with correct url', () => {
        const url = `${API_TRACKS_URL}?${CLIENT_ID_PARAM}`;

        server.respondWith('get', url, request => {
          expect(request.url).toBe(url);
          return successResponse;
        });

        api.fetch(url);
        server.respond();
      });
    });


    describe('api.fetchSearchResults()', () => {
      xit('should perform GET request with correct url', () => {
        const url = `${API_TRACKS_URL}?${CLIENT_ID_PARAM}&${PAGINATION_PARAMS}&${queryParam}`;

        server.respondWith('get', url, request => {
          expect(request.url).toBe(url);
          return successResponse;
        });

        api.fetchSearchResults(queryValue);
        server.respond();
      });
    });


    describe('api.fetchUser()', () => {
      xit('should perform GET request with correct url', () => {
        const userId = 123;
        const url = `${API_USERS_URL}/${userId}?${CLIENT_ID_PARAM}`;

        server.respondWith('get', url, request => {
          expect(request.url).toBe(url);
          return successResponse;
        });

        api.fetchUser(userId);
        server.respond();
      });
    });


    describe('api.fetchUserLikes()', () => {
      xit('should perform GET request with correct url', () => {
        const userId = 123;
        const url = `${API_USERS_URL}/${userId}/favorites?${CLIENT_ID_PARAM}&${PAGINATION_PARAMS}`;

        server.respondWith('get', url, request => {
          expect(request.url).toBe(url);
          return successResponse;
        });

        api.fetchUserLikes(userId);
        server.respond();
      });
    });


    describe('api.fetchUserTracks()', () => {
      xit('should perform GET request with correct url', () => {
        const userId = 123;
        const url = `${API_USERS_URL}/${userId}/tracks?${CLIENT_ID_PARAM}&${PAGINATION_PARAMS}`;

        server.respondWith('get', url, request => {
          expect(request.url).toBe(url);
          return successResponse;
        });

        api.fetchUserTracks(userId);
        server.respond();
      });
    });
  });
});
