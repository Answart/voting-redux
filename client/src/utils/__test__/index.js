export {
  mockUser, mockPoll, mockPolls,
  mockUsersInitialState, mockPollsInitialState
} from './mockState';
export {
  mockInitialStore, mockStore,
  routerContext, context
} from './mockStore';
export {
  asyncFlush,
  click, submit,
  clickLink, clickButton, submitButton,
  setValue, fillInput, fillFormInput, fillInputValue, getInput,
  createMount, createShallow,
  muiMounter,
  MuiMountWithContext, MuiShallowWithContext,
  mountConnected,
  mountWithRouter, shallowWithRouter,
  mountWithRouterConnected, shallowWithRouterConnected
} from './test.helper';
