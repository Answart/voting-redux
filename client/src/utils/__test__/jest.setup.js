import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson, { mountToJson } from 'enzyme-to-json';


// ENZYME support for REACT 16 not complete. (re: rerender on prop changes)
// https://github.com/airbnb/enzyme/issues/1229
// https://github.com/airbnb/enzyme/issues/1553

configure({ adapter: new Adapter() });


// mock popper.js to calm jest-jasmine2 "document.createRange is not a function" error
jest.mock('popper.js', () => {
  const PopperJS = jest.requireActual('popper.js');

  return class {
    static placements = PopperJS.placements;

    constructor() {
      return {
        destroy: () => {},
        scheduleUpdate: () => {}
      };
    }
  };
});

var localStorageMock = (function() {
  var store = {};
  return {
    getItem: function(key) {
      return store[key] || null;
    },
    setItem: function(key, value) {
      store[key] = value.toString();
    },
    removeItem: function(key) {
      delete store[key];
    },
    clear: function() {
      store = {};
    }
  };
})();


// ==========================================================
// GLOBAL MOCKS

// Make Enzyme functions available in all test files without importing
global.localStorage = localStorageMock;
// SNAPSHOOTING
global.toJson = toJson;
global.mountToJson = mountToJson;
