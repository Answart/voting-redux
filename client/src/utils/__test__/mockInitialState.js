import { pollReducer } from '../../core/polls';
import { userReducer } from '../../core/users';
import { reducer as form } from 'redux-form';


// ======================================================
// INITIAL STATES

const usersInitialState = {
  authedUser: {
    loading: false,
    error: null,
    message: null,
    token: null,
    user: null
  }
};
const pollsInitialState = {
  all: {
    loading: false, error: null, polls: null
  },
  filtered: {
    loading: false, error: null, message: null, filters: null, polls: null
  },
  active: {
    loading: false, error: null, message: null, poll: null
  },
  viewed: {
    loading: false, error: null, message: null, id: null, poll: null
  }
};


// ======================================================
// MOCK SCENARIOS

const users = userReducer(usersInitialState, { type: 'AUTH_USER_SUCCESS', message: 'Yay', user: {
  name: 'alexandra',
  cuid: '12345',
  activity: [],
  token: '54321'
}});

const allPolls = [{
    cuid: '1',
    title: 'random title',
    user_name: 'name1',
    votes: 10,
    open: true,
    date_created: '2018-04-06T04:34:25.183Z',
    choices: [
      { id: 0, label: 'red', vote: 4 },
      { id: 1, label: 'blue', vote: 6 }
    ]
  }, {
    cuid: '2',
    title: 'random title2',
    user_name: 'name2',
    votes: 30,
    open: false,
    date_created: '2018-04-06T04:34:01.247Z',
    choices: [
      { id: 0, label: 'red2', vote: 14 },
      { id: 1, label: 'blue2', vote: 16 }
    ]
  }, {
    cuid: '3',
    title: 'random title3',
    user_name: 'name3',
    votes: 30,
    open: false,
    date_created: '2018-04-06T04:34:01.247Z',
    choices: [
      { id: 0, label: 'red3', vote: 14 },
      { id: 1, label: 'blue3', vote: 16 }
    ]
  }, {
    cuid: '4',
    title: 'random title4',
    user_name: 'name4',
    votes: 30,
    open: false,
    date_created: '2018-04-06T04:34:01.247Z',
    choices: [
      { id: 0, label: 'red2', vote: 14 },
      { id: 1, label: 'blue2', vote: 16 }
    ]
  }, {
    cuid: '5',
    title: 'random title5',
    user_name: 'name5',
    votes: 30,
    open: false,
    date_created: '2018-04-06T04:34:01.247Z',
    choices: [
      { id: 0, label: 'red2', vote: 14 },
      { id: 1, label: 'blue2', vote: 16 }
    ]
  }, {
    cuid: '6',
    title: 'random title6',
    user_name: 'name6',
    votes: 30,
    open: false,
    date_created: '2018-04-06T04:34:01.247Z',
    choices: [
      { id: 0, label: 'red2', vote: 14 },
      { id: 1, label: 'blue2', vote: 16 }
    ]
  }, {
    cuid: '7',
    title: 'random title7',
    user_name: 'name7',
    votes: 30,
    open: false,
    date_created: '2018-04-06T04:34:01.247Z',
    choices: [
      { id: 0, label: 'red2', vote: 14 },
      { id: 1, label: 'blue2', vote: 16 }
    ]
  }, {
    cuid: '8',
    title: 'random title8',
    user_name: 'name8',
    votes: 30,
    open: false,
    date_created: '2018-04-06T04:34:01.247Z',
    choices: [
      { id: 0, label: 'red2', vote: 14 },
      { id: 1, label: 'blue2', vote: 16 }
    ]
  }, {
    cuid: '9',
    title: 'random title9',
    user_name: 'name9',
    votes: 30,
    open: false,
    date_created: '2018-04-06T04:34:01.247Z',
    choices: [
      { id: 0, label: 'red2', vote: 14 },
      { id: 1, label: 'blue2', vote: 16 }
    ]
  }, {
    cuid: '10',
    title: 'random title10',
    user_name: 'name10',
    votes: 30,
    open: false,
    date_created: '2018-04-06T04:34:01.247Z',
    choices: [
      { id: 0, label: 'red2', vote: 14 },
      { id: 1, label: 'blue2', vote: 16 }
    ]
  }
];
pollsInitialState.viewed.id = '2';
pollsInitialState.filtered.filters = [{
  label: 'User',
  key: 'user_name',
  value: 'name2'
}];
const polls = pollReducer(pollsInitialState, { type: 'GET_POLLS_SUCCESS', polls: allPolls });


// ======================================================
// MOCK STATES

export const mockInitialState = { form, users: usersInitialState, polls: pollsInitialState };
export const mockState = { form, users, polls };
