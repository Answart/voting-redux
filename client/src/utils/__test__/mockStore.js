import PropTypes from 'prop-types';
import createSagaMiddleware from 'redux-saga';
import configureStore from 'redux-mock-store';
import createRouterContext from 'react-router-test-context';
import { mockInitialState, mockState } from './mockState';

const sagaMiddleware = createSagaMiddleware();
const configMockStore = configureStore([sagaMiddleware]);


// ==========================================================
// MOCK CONTEXTS

const mockFn = jest.fn;
export const context = {
  context: {
    // muiTheme,
    router: {
      isActive: mockFn,
      push: mockFn,
      replace: mockFn,
      go: mockFn,
      goBack: mockFn,
      goForward: mockFn,
      setRouteLeaveHook: mockFn,
      createHref: mockFn,
    }
  },
  childContextTypes: {
    router: PropTypes.object.isRequired,
    muiTheme: PropTypes.object
  }
};
export const routerContext = createRouterContext();


// ==========================================================
// MOCK STORES

export const mockInitialStore = configMockStore(mockInitialState);
export const mockStore = configMockStore(mockState);
