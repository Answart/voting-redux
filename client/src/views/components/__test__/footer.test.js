import React from 'react';
import { muiMounter } from '../../../utils/__test__';
// Import components
import Footer from '../footer';


describe('<Footer />', () => {
  it('renders properly', () => {
    let wrapper = muiMounter(<Footer />);
    expect(wrapper).toBeDefined();
    const icons = wrapper.find('IconButton');
    expect(icons.length).toBe(3);
    expect(icons.get(0).props.href).toBe('https://github.com/answart');
    expect(icons.get(1).props.href).toBe('https://www.linkedin.com/in/alexandra-swart-582a1235/');
    expect(icons.get(2).props.href).toBe('mailto:answart@sbcglobal.net.com');
    expect(wrapper.find('p').text()).toBe('Â© Alexandra Swart 2018')
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });
});
