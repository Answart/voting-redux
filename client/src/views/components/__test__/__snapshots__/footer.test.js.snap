// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders properly 1`] = `
<Footer>
  <footer
    id="footer"
  >
    <WithStyles(IconButton)
      color="primary"
      href="https://github.com/answart"
    >
      <IconButton
        classes={
          Object {
            "colorInherit": "MuiIconButton-colorInherit-2",
            "colorPrimary": "MuiIconButton-colorPrimary-3",
            "colorSecondary": "MuiIconButton-colorSecondary-4",
            "disabled": "MuiIconButton-disabled-5",
            "label": "MuiIconButton-label-6",
            "root": "MuiIconButton-root-1",
          }
        }
        color="primary"
        disabled={false}
        href="https://github.com/answart"
      >
        <WithStyles(ButtonBase)
          centerRipple={true}
          className="MuiIconButton-root-1 MuiIconButton-colorPrimary-3"
          disabled={false}
          focusRipple={true}
          href="https://github.com/answart"
        >
          <ButtonBase
            centerRipple={true}
            className="MuiIconButton-root-1 MuiIconButton-colorPrimary-3"
            classes={
              Object {
                "disabled": "MuiButtonBase-disabled-8",
                "focusVisible": "MuiButtonBase-focusVisible-9",
                "root": "MuiButtonBase-root-7",
              }
            }
            component="button"
            disableRipple={false}
            disableTouchRipple={false}
            disabled={false}
            focusRipple={true}
            href="https://github.com/answart"
            tabIndex="0"
            type="button"
          >
            <a
              className="MuiButtonBase-root-7 MuiIconButton-root-1 MuiIconButton-colorPrimary-3"
              href="https://github.com/answart"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                className="MuiIconButton-label-6"
              >
                <span
                  className="fa fa-github icon"
                />
              </span>
              <NoSsr
                defer={false}
                fallback={null}
              >
                <WithStyles(TouchRipple)
                  center={true}
                  innerRef={[Function]}
                >
                  <TouchRipple
                    center={true}
                    classes={
                      Object {
                        "child": "MuiTouchRipple-child-14",
                        "childLeaving": "MuiTouchRipple-childLeaving-15",
                        "childPulsate": "MuiTouchRipple-childPulsate-16",
                        "ripple": "MuiTouchRipple-ripple-11",
                        "ripplePulsate": "MuiTouchRipple-ripplePulsate-13",
                        "rippleVisible": "MuiTouchRipple-rippleVisible-12",
                        "root": "MuiTouchRipple-root-10",
                      }
                    }
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      className="MuiTouchRipple-root-10"
                      component="span"
                      enter={true}
                      exit={true}
                    >
                      <span
                        className="MuiTouchRipple-root-10"
                      />
                    </TransitionGroup>
                  </TouchRipple>
                </WithStyles(TouchRipple)>
              </NoSsr>
            </a>
          </ButtonBase>
        </WithStyles(ButtonBase)>
      </IconButton>
    </WithStyles(IconButton)>
    <WithStyles(IconButton)
      color="primary"
      href="https://www.linkedin.com/in/alexandra-swart-582a1235/"
    >
      <IconButton
        classes={
          Object {
            "colorInherit": "MuiIconButton-colorInherit-2",
            "colorPrimary": "MuiIconButton-colorPrimary-3",
            "colorSecondary": "MuiIconButton-colorSecondary-4",
            "disabled": "MuiIconButton-disabled-5",
            "label": "MuiIconButton-label-6",
            "root": "MuiIconButton-root-1",
          }
        }
        color="primary"
        disabled={false}
        href="https://www.linkedin.com/in/alexandra-swart-582a1235/"
      >
        <WithStyles(ButtonBase)
          centerRipple={true}
          className="MuiIconButton-root-1 MuiIconButton-colorPrimary-3"
          disabled={false}
          focusRipple={true}
          href="https://www.linkedin.com/in/alexandra-swart-582a1235/"
        >
          <ButtonBase
            centerRipple={true}
            className="MuiIconButton-root-1 MuiIconButton-colorPrimary-3"
            classes={
              Object {
                "disabled": "MuiButtonBase-disabled-8",
                "focusVisible": "MuiButtonBase-focusVisible-9",
                "root": "MuiButtonBase-root-7",
              }
            }
            component="button"
            disableRipple={false}
            disableTouchRipple={false}
            disabled={false}
            focusRipple={true}
            href="https://www.linkedin.com/in/alexandra-swart-582a1235/"
            tabIndex="0"
            type="button"
          >
            <a
              className="MuiButtonBase-root-7 MuiIconButton-root-1 MuiIconButton-colorPrimary-3"
              href="https://www.linkedin.com/in/alexandra-swart-582a1235/"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                className="MuiIconButton-label-6"
              >
                <span
                  className="fa fa-linkedin-square icon"
                />
              </span>
              <NoSsr
                defer={false}
                fallback={null}
              >
                <WithStyles(TouchRipple)
                  center={true}
                  innerRef={[Function]}
                >
                  <TouchRipple
                    center={true}
                    classes={
                      Object {
                        "child": "MuiTouchRipple-child-14",
                        "childLeaving": "MuiTouchRipple-childLeaving-15",
                        "childPulsate": "MuiTouchRipple-childPulsate-16",
                        "ripple": "MuiTouchRipple-ripple-11",
                        "ripplePulsate": "MuiTouchRipple-ripplePulsate-13",
                        "rippleVisible": "MuiTouchRipple-rippleVisible-12",
                        "root": "MuiTouchRipple-root-10",
                      }
                    }
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      className="MuiTouchRipple-root-10"
                      component="span"
                      enter={true}
                      exit={true}
                    >
                      <span
                        className="MuiTouchRipple-root-10"
                      />
                    </TransitionGroup>
                  </TouchRipple>
                </WithStyles(TouchRipple)>
              </NoSsr>
            </a>
          </ButtonBase>
        </WithStyles(ButtonBase)>
      </IconButton>
    </WithStyles(IconButton)>
    <WithStyles(IconButton)
      color="primary"
      href="mailto:answart@sbcglobal.net.com"
    >
      <IconButton
        classes={
          Object {
            "colorInherit": "MuiIconButton-colorInherit-2",
            "colorPrimary": "MuiIconButton-colorPrimary-3",
            "colorSecondary": "MuiIconButton-colorSecondary-4",
            "disabled": "MuiIconButton-disabled-5",
            "label": "MuiIconButton-label-6",
            "root": "MuiIconButton-root-1",
          }
        }
        color="primary"
        disabled={false}
        href="mailto:answart@sbcglobal.net.com"
      >
        <WithStyles(ButtonBase)
          centerRipple={true}
          className="MuiIconButton-root-1 MuiIconButton-colorPrimary-3"
          disabled={false}
          focusRipple={true}
          href="mailto:answart@sbcglobal.net.com"
        >
          <ButtonBase
            centerRipple={true}
            className="MuiIconButton-root-1 MuiIconButton-colorPrimary-3"
            classes={
              Object {
                "disabled": "MuiButtonBase-disabled-8",
                "focusVisible": "MuiButtonBase-focusVisible-9",
                "root": "MuiButtonBase-root-7",
              }
            }
            component="button"
            disableRipple={false}
            disableTouchRipple={false}
            disabled={false}
            focusRipple={true}
            href="mailto:answart@sbcglobal.net.com"
            tabIndex="0"
            type="button"
          >
            <a
              className="MuiButtonBase-root-7 MuiIconButton-root-1 MuiIconButton-colorPrimary-3"
              href="mailto:answart@sbcglobal.net.com"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                className="MuiIconButton-label-6"
              >
                <span
                  className="fa fa-envelope icon"
                />
              </span>
              <NoSsr
                defer={false}
                fallback={null}
              >
                <WithStyles(TouchRipple)
                  center={true}
                  innerRef={[Function]}
                >
                  <TouchRipple
                    center={true}
                    classes={
                      Object {
                        "child": "MuiTouchRipple-child-14",
                        "childLeaving": "MuiTouchRipple-childLeaving-15",
                        "childPulsate": "MuiTouchRipple-childPulsate-16",
                        "ripple": "MuiTouchRipple-ripple-11",
                        "ripplePulsate": "MuiTouchRipple-ripplePulsate-13",
                        "rippleVisible": "MuiTouchRipple-rippleVisible-12",
                        "root": "MuiTouchRipple-root-10",
                      }
                    }
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      className="MuiTouchRipple-root-10"
                      component="span"
                      enter={true}
                      exit={true}
                    >
                      <span
                        className="MuiTouchRipple-root-10"
                      />
                    </TransitionGroup>
                  </TouchRipple>
                </WithStyles(TouchRipple)>
              </NoSsr>
            </a>
          </ButtonBase>
        </WithStyles(ButtonBase)>
      </IconButton>
    </WithStyles(IconButton)>
    <p>
      © Alexandra Swart 2018
    </p>
  </footer>
</Footer>
`;
